### ISSUE TITLE
Questions about training with DIP-IMU dataset

### ISSUE THREAD

**Issue by huzijun1996** [Xinyu-Yi/PIP#47]
Hi, I'm having some problems with the reproduction training process and hope I can get some help from you.

1、Regarding the rotation information we used in the DIP-IMU training and the position information we synthesized subsequently. Could we just use the “gt” data in the DIP-IMU dataset to synthesize all the data we use for training, regardless of “imu_ori” and “imu_acc”?
2、The DIP-IMU dataset doesn't have translation data, so can we train Trans-B1(Va) and Trans-B2(Cf)? Or is it only used to train the first three networks?
3、You said "Maybe you can check the estimated velocity v.s. the ground truth" in (https://github.com/Xinyu-Yi/PIP/issues/11), isn't it true that we can just treat the position difference as velocity without divide by the time interval delta_t? From my training results, I find that divide by delta_t doesn't rather reasonable for the training result.
4、You said in (https://github.com/Xinyu-Yi/TransPose/issues/65) that the joint positions generated by forward kinematics in DIP-IMU and TotalCapture need to be rotated by left-multiplying the inverse root rotation, so do we need to do the same for the joint.pt generated from the AMASS dataset in order to train it?
![Image](https://github.com/user-attachments/assets/bf82beb3-6b0b-4a77-a68e-a8b175d46a90)

**Comment by Xinyu-Yi** [comment]
1. We use DIP-IMU for real IMU measurements for fine-tuning the model. So I think you'd better use the real IMU data for input to learn the real noise patterns.
2. Yes, DIP-IMU is only used to fine-tune the first 3 pose networks.
3. Yes, you can define the velocity as the position difference (without dividing by the delta_t). But no matter how you scale it, you need to transform it to the root frame during training (left-multiply the root rotation's inverse) as all inputs are root-frame measurements.
4. Exactly. First minus the root position, then left-multiply the root rotation inverse. i.e., we use root-frame joint positions.

**Comment by huzijun1996** [comment]
Thank you for your answer.
Regarding question 3, I have other questions:
1. Is it necessary to do normalization during training and testing (including AMASS, DIP-IMU, TotalCapture), i.e. dividing by acc_scale=30 and vel_scale=3 as you have given in your example code? I noticed that you use vel_scale in both the PIP and Transpose code, but one of them uses acc_scale and the other doesn't, why is that?
2. If I don't divide by delta_t during training, do I have to divide by delta_t during actual use (i.e. 60)? I noticed that you use it differently in PIP and transpose, one divides by 60 and one doesn't.

In addition, when using the DIP-IMU dataset for fine-tuning, you had mentioned that we cannot let the fine-tuning converge, and that several epochs are enough (https://github.com/Xinyu-Yi/PIP/issues/34):
1. How do you determine that the fine-tuning has been done?
2. Do I need to fine-tune all three pose networks together, or is it the same as during training, one by one?

**Comment by Xinyu-Yi** [comment]
1. I use the scale to make the input and output numbers in [-3, 3]. But later I find that it does not add much accuracy. Without normalization the training can also converge. So I remove them in later works.
2. As long as you make the training and evaluation consistent.
3. I finetune it with one epoch. I also tried adding the entire DIP-IMU into the entire AMASS dataset (mix them for training, and only train once) and the results are also good and even better.
4. One by one.

I think for PIP, you can easily outperform my reported results if you more carefully tune the hyperparameters and training method. You may not strictly follow my paper as it is not the optimal choice.

**Comment by huzijun1996** [comment]
Thank you for your reply. After my experiments this week, I still have some problems in training the synthetic dataset of RNN4 and RNN5, my training result is not very good, I hope to get your guidance:

1. According to your answer above, which means all the speeds should be [-10,10] or [-9, 9] before you use the scale(vel_scale). But I used the joint in preprocess.py, combined with the formula in Appendix A in the PIP (I didn't multiply the root inverse matrix, because I noticed that you multiplied it additionally only during the test). With all my default initial velocities at 0, I calculated the TotalCapture velocity data to be [-23,23],and the AMASS velocity data got [-36,32], could you advice me if I'm using it incorrectly?
![Image](https://github.com/user-attachments/assets/b5ac809d-36a0-472d-80f3-da92141826cb)

2. I followed the method you mentioned in Transpose to set the foot-ground contact probability for ground truth. But in the process of using it, I got different results from the weights you provided, using provided weights, the results of RNN5 will all be far beyond the range of [0,1], while my own weights are most of the time in the range of [0,1] (I didn't use sigmoid during training), is there any problem with my synthesis?
![Image](https://github.com/user-attachments/assets/613b5cc8-aeef-46b0-9436-49cb3c060634)
![Image](https://github.com/user-attachments/assets/7194197d-2248-4e12-88ff-f0c6c4bcd59a)
![Image](https://github.com/user-attachments/assets/50b8349f-8536-4d2c-8756-30dfb6614c5f)

3. May I ask in what ratio you divided the train set and validation set of the dataset for training and fine-tuning with AMASS and DIP-IMU? (I am using 8:2 ratio).

**Comment by Xinyu-Yi** [comment]
1. I think it is OK. vel scale and acc scale are just approximations and without them you should also get similar results
2. The output can be large (without sigmoid), and the label are 0 or 1. Then you can use BCELossWithLogits
3. I remember that in this work I follow the AMASS official recommendation for the validation split and discard the test split.

**Comment by kevin-nino** [comment]
@huzijun1996 Hi! I’m currently trying to train the TransPose network, but I haven’t been able to achieve good results so far. If possible, could you share some guidance or the code you used for training? It would help me a lot.

You can also reach me at kevin.nino@upr.edu; Thanks in advance!
